### PR TITLE
Refresh plugin for October 2023

### PR DESCRIPTION
**Pull request by basil**
See #194

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
+  <extension>
+    <groupId>io.jenkins.tools.incrementals</groupId>
+    <artifactId>git-changelist-maven-extension</artifactId>
+    <version>1.7</version>
+  </extension>
+</extensions>
```
Comment (basil) [inline .mvn/extensions.xml:7]: As in the archetype.

File: .mvn/maven.config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-Pconsume-incrementals
+-Pmight-produce-incrementals
```
Comment (basil) [inline .mvn/maven.config:2]: As in the archetype.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
 buildPlugin(useContainerAgent: true, configurations: [
-  [platform: 'linux', jdk: 11],
-  [platform: 'windows', jdk: 11],
+  [platform: 'linux', jdk: 21],
+  [platform: 'windows', jdk: 17],
```
Comment (basil) [inline Jenkinsfile:3]: See #194.

File: package.json [hunk 4]
```diff
@@ -9,7 +9,8 @@
     "README.md"
   ],
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "mvnbuild": "gulp bundle",
+    "mvntest": "gulp test"
```
Comment (basil) [inline package.json:13]: Newer version of NPM complains about this duplicate key. The second version was the one actually being used anyway.

File: pom.xml [hunk 5]
```diff
@@ -4,31 +4,34 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.66</version>
+    <version>4.74</version>
```
Comment (basil) [inline pom.xml:7]: Latest at the time of this writing.

File: pom.xml [hunk 6]
```diff
@@ -4,31 +4,34 @@
...
     <relativePath />
   </parent>
   <artifactId>sse-gateway</artifactId>
-  <version>1.27-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
```
Comment (basil) [inline pom.xml:11]: Incrementalified.

File: pom.xml [hunk 7]
```diff
@@ -4,31 +4,34 @@
...
   <packaging>hpi</packaging>
 
   <properties>
-    <jenkins.version>2.361</jenkins.version>
-    <node.version>5.8.0</node.version>
-    <npm.version>3.7.3</npm.version>
+    <revision>1.27</revision>
+    <changelist>-SNAPSHOT</changelist>
```
Comment (basil) [inline pom.xml:16]: Incrementalified.

File: pom.xml [hunk 8]
```diff
@@ -4,31 +4,34 @@
...
+    <jenkins.version>2.387.3</jenkins.version>
```
Comment (basil) [inline pom.xml:17]: https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/

File: pom.xml [hunk 9]
```diff
@@ -4,31 +4,34 @@
...
+    <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
```
Comment (basil) [inline pom.xml:18]: Incrementalified.

File: pom.xml [hunk 10]
```diff
@@ -4,31 +4,34 @@
...
+    <node.version>8.17.0</node.version>
+    <npm.version>6.13.4</npm.version>
```
Comment (basil) [inline pom.xml:20]: Late 2019 vintage; should hopefully reduce BOM/PCT flakiness.

File: pom.xml [hunk 11]
```diff
@@ -4,31 +4,34 @@
...
     <localRepoPath>${project.build.directory}/local-repo</localRepoPath>
     <settingsPath>src/it/settings.xml</settingsPath>
     <maven.dependency.plugin.version>3.1.1</maven.dependency.plugin.version>
   </properties>
 
   <name>Server Sent Events (SSE) Gateway Plugin</name>
   <description>Server Sent Events (SSE) Gateway Plugin</description>
-  <url>https://github.com/jenkinsci/sse-gateway-plugin</url>
+  <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```
Comment (basil) [inline pom.xml:28]: As in the archetype.

File: pom.xml [hunk 12]
```diff
@@ -4,31 +4,34 @@
...
 
   <scm>
-    <connection>scm:git:https://github.com/jenkinsci/sse-gateway-plugin.git</connection>
-    <developerConnection>scm:git:https://github.com/jenkinsci/sse-gateway-plugin.git</developerConnection>
-    <url>https://github.com/jenkinsci/sse-gateway-plugin</url>
-    <tag>HEAD</tag>
+    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
+    <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
+    <url>https://github.com/${gitHubRepo}</url>
+    <tag>${scmTag}</tag>
```
Comment (basil) [inline pom.xml:34]: Incrementalified.

File: pom.xml [hunk 13]
```diff
@@ -55,26 +58,18 @@
...
+        <version>2507.vcb_18c56b_f57c</version>
         <scope>import</scope>
         <type>pom</type>
       </dependency>
-      <dependency>
-        <groupId>junit</groupId>
-        <artifactId>junit</artifactId>
-        <version>4.13.2</version>
-        <scope>test</scope>
-      </dependency>
```
Comment (basil) [inline pom.xml:69]: Managed in the plugin parent POM.

File: pom.xml [hunk 14]
```diff
@@ -55,26 +58,18 @@
...
     </dependencies>
   </dependencyManagement>
 
   <dependencies>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>pubsub-light</artifactId>
-      <version>1.17</version>
```
Comment (basil) [inline pom.xml:77]: Managed in the plugin BOM.

File: pom.xml [hunk 15]
```diff
@@ -86,7 +81,7 @@
     <dependency>
       <groupId>com.github.eirslett</groupId>
       <artifactId>frontend-plugin-core</artifactId>
-      <version>1.12.1</version>
+      <version>${frontend-version}</version>
```
Comment (basil) [inline pom.xml:84]: Managed in the plugin parent POM.

File: pom.xml [hunk 16]
```diff
@@ -142,11 +137,6 @@
           -->
           <version>2.7</version>
         </plugin>
-        <plugin>
-          <groupId>org.codehaus.mojo</groupId>
-          <artifactId>findbugs-maven-plugin</artifactId>
-          <version>3.0.5</version>
-        </plugin>
```
Comment (basil) [inline pom.xml:149]: No longer needed.

**Comment by MarkEWaite** [comment]
@olamy , @car-roll , or @dwnusbaum could you review and merge this?  We are announcing Java 21 support in Jenkins 2.426.1 on 15 Nov 2023 and would like most of the top 250 plugins to already be testing with Java 21.  One of the benefits of this pull request is that it implements Java 21 and Java 17 support.

**Comment by olamy** [comment]
@MarkEWaite the PR includes a core requirement bump. Not sure if it's really important for this plugin to require to have it mentioned in the PR title and not having the label `skip-changelog` so this will be part of the release notes.
wdyt?

**Comment by olamy** [comment]
ok let's say it's not important

**Comment by MarkEWaite** [comment]
I like mentioning core requirement bumps in PR titles for my own convenience.  It helps me identify transition points in the plugin release history.  You're welcome to update the title of the pull request to mention that it now requires Jenkins 2.,387.3.

I have been using `skip-changelog` less and less with plugin updates.  I think you're right that it would be good to remove the `skip-changelog` label from this pull request.

**Pull request merged**